[inductor] Enable Mypy Checking for torch/_inductor/codecache.py

masnesral opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #108789

Summary: Add type annotations to torch/_inductor/codecache.py and enable mypy checking

Test Plan:
`lintrunner torch/_inductor/*.py`
`python test/inductor/test_max_autotune.py`
`python test/inductor/test_aot_inductor.py`
`python test/inductor/test_torchinductor.py`

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

masnesral committed:
[inductor] Enable Mypy Checking for torch/_inductor/codecache.py

Summary: Add type annotations to torch/_inductor/codecache.py and enable mypy checking

Test Plan:
`python test/inductor/test_max_autotune.py`
`python test/inductor/test_aot_inductor.py`
`python test/inductor/test_torchinductor.py`

[ghstack-poisoned]

masnesral commented on torch/_inductor/codecache.py:
```diff
@@ -119,15 +129,15 @@ def cpp_wrapper_cache_dir(name):
     @staticmethod
     @functools.lru_cache(None)
-    def get_system():
+    def get_system() -> Dict[str, Any]:
```
So there are several places throughout where I've used `Any` and would have liked to avoid. But a lot of these caching utilities generally seem to be pretty liberal about the types of objects, e.g., I was seeing instances of both objects and primitive types. So I kinda gave up wherever you see `Any`. Let me know if anywhere looks especially lazy and/or you'd prefer to be stricter even if it means a ridiculous Union of possible types.

Skylion007 replied:
Should it be Any or just Python object? Main difference is Any can be a None type too etc.

masnesral commented on torch/_inductor/codecache.py:
```diff
@@ -392,26 +405,26 @@ def get_current_callable(self):
             compiled_graph.artifact_path,
-            compiled_graph.cache_linemap
-            if compiled_graph.cache_linemap is not None
-            else (),
```
See below for change to `load_by_key_path` where the default is now None.

masnesral commented on torch/_inductor/codecache.py:
```diff
@@ -1126,27 +1152,39 @@ def load(cls, source_code):
+        cls,
+        key: str,
+        path: str,
+        linemap: Optional[List[Tuple[int, Optional[str]]]] = None,
```
The default was previously an empty tuple, but the linemap is actually a list. That happened to work because the `zip` usage "did the right" thing and created an empty list when operating on an empty tuple. But I changed this to a real list and then obeyed the linter when it (correctly) doesn't want an empty list as a default param.

masnesral committed:
Update on "[inductor] Enable Mypy Checking for torch/_inductor/codecache.py"


Summary: Add type annotations to torch/_inductor/codecache.py and enable mypy checking

Test Plan:
`python test/inductor/test_max_autotune.py`
`python test/inductor/test_aot_inductor.py`
`python test/inductor/test_torchinductor.py`

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

masnesral committed:
Update on "[inductor] Enable Mypy Checking for torch/_inductor/codecache.py"


Summary: Add type annotations to torch/_inductor/codecache.py and enable mypy checking

Test Plan:
`lintrunner torch/_inductor/*.py`
`python test/inductor/test_max_autotune.py`
`python test/inductor/test_aot_inductor.py`
`python test/inductor/test_torchinductor.py`

[ghstack-poisoned]

masnesral commented on torch/_inductor/codecache.py:
```diff
@@ -1253,9 +1293,9 @@ def load(cls, source_code, func_name, key, cuda):
                 else:
                     log.debug("Found target .so, cpp wrapper loading %s", filepath)
-                    spec = importlib.util.spec_from_file_location(name, filepath)
+                    spec = importlib.util.spec_from_file_location(name, filepath)  # type: ignore[attr-defined]
```
The linter is telling me that importlib.util doesn't even exist. So.... maybe this code path is not being exercised any more?

Skylion007 commented on torch/_inductor/codecache.py:
```diff
@@ -873,22 +890,22 @@ def get_include_and_linking_paths(
-    lpaths = " ".join(["-L" + p for p in lpaths])
-    libs = " ".join(["-l" + p for p in libs])
-    return ipaths, lpaths, libs, macros
+    ipaths_str = " ".join(["-I" + p for p in ipaths])
```
While you are touching this, it would be nice if these were generators instead of list comprehensions.

Skylion007 commented on torch/_inductor/codecache.py:
```diff
@@ -612,7 +625,8 @@ class InvalidVecISA(VecISA):
         return "INVALID_VEC_ISA"
 
-    def __bool__(self):
+    @functools.lru_cache(None)
```
This is bug prone behavior to LRU cache on methods (attempts to cache self which is often bad). I really need to enable that flake8-bugprone rule.

Skylion007 reviewed (approved):
Minor nits, but LGTM.

masnesral committed:
nits on "[inductor] Enable Mypy Checking for torch/_inductor/codecache.py"


Summary: Add type annotations to torch/_inductor/codecache.py and enable mypy checking

Test Plan:
`lintrunner torch/_inductor/*.py`
`python test/inductor/test_max_autotune.py`
`python test/inductor/test_aot_inductor.py`
`python test/inductor/test_torchinductor.py`

[ghstack-poisoned]

eellison reviewed (approved):
thanks, looks great !

masnesral commented:
@pytorchbot merge

masnesral committed:
rebase on "[inductor] Enable Mypy Checking for torch/_inductor/codecache.py"


Summary: Add type annotations to torch/_inductor/codecache.py and enable mypy checking

Test Plan:
`lintrunner torch/_inductor/*.py`
`python test/inductor/test_max_autotune.py`
`python test/inductor/test_aot_inductor.py`
`python test/inductor/test_torchinductor.py`

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

masnesral commented:
@pytorchbot merge

masnesral committed:
rebase on "[inductor] Enable Mypy Checking for torch/_inductor/codecache.py"


Summary: Add type annotations to torch/_inductor/codecache.py and enable mypy checking

Test Plan:
`lintrunner torch/_inductor/*.py`
`python test/inductor/test_max_autotune.py`
`python test/inductor/test_aot_inductor.py`
`python test/inductor/test_torchinductor.py`

[ghstack-poisoned]

masnesral commented:
@pytorchbot merge

masnesral committed:
rebase on "[inductor] Enable Mypy Checking for torch/_inductor/codecache.py"


Summary: Add type annotations to torch/_inductor/codecache.py and enable mypy checking

Test Plan:
`lintrunner torch/_inductor/*.py`
`python test/inductor/test_max_autotune.py`
`python test/inductor/test_aot_inductor.py`
`python test/inductor/test_torchinductor.py`

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

masnesral committed:
rebase on "[inductor] Enable Mypy Checking for torch/_inductor/codecache.py"


Summary: Add type annotations to torch/_inductor/codecache.py and enable mypy checking

Test Plan:
`lintrunner torch/_inductor/*.py`
`python test/inductor/test_max_autotune.py`
`python test/inductor/test_aot_inductor.py`
`python test/inductor/test_torchinductor.py`

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

masnesral committed:
rebase on "[inductor] Enable Mypy Checking for torch/_inductor/codecache.py"


Summary: Add type annotations to torch/_inductor/codecache.py and enable mypy checking

Test Plan:
`lintrunner torch/_inductor/*.py`
`python test/inductor/test_max_autotune.py`
`python test/inductor/test_aot_inductor.py`
`python test/inductor/test_torchinductor.py`

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

masnesral committed:
rm unused imports on "[inductor] Enable Mypy Checking for torch/_inductor/codecache.py"


Summary: Add type annotations to torch/_inductor/codecache.py and enable mypy checking

Test Plan:
`lintrunner torch/_inductor/*.py`
`python test/inductor/test_max_autotune.py`
`python test/inductor/test_aot_inductor.py`
`python test/inductor/test_torchinductor.py`

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

masnesral commented:
@pytorchbot merge
